### ISSUE TITLE
Is e.g. `path.separator` part of the public API?

### ISSUE THREAD

**Issue by ericphanson** [rofinn/FilePathsBase.jl#131]
`getproperty` is defined but not `propertynames` and it's unclear if downstream code should use it or not. So the question goes for all the `getproperty` overloads.

**Comment by rofinn** [comment]
I think at this point it is safe to assume all path types will have a `fp.separator` and `fp.segments`. I considered making individual accessors, but that seemed a bit verbose.

**Comment by ericphanson** [comment]
Do you think we should define `propertynames` then?

**Comment by rofinn** [comment]
Sure, that seems reasonable. I didn't actually realize `propertynames` has `private=false` for this intended purpose. The idea of public vs private in Julia has always seemed a bit vague to me.

**Comment by ericphanson** [comment]
Yeah, I haven't seen the `private` bit of propertynames used much in the wild, but I think it could have a useful role in communicating what's public or not.
